process-analyser/src_hd/com/axonivy/utils/process/analyser/component/CustomFilter/CustomFilter.xhtml

```diff
@@ -4,89 +4,94 @@
 
           <!-- STRING, TEXT -->
-           <h:panelGroup layout="block" rendered="#{processesAnalyticsBean.filterDropdownVisible and 
-                ('STRING' == fieldEntry.key.customFieldMeta.type() or 
-                 'TEXT' == fieldEntry.key.customFieldMeta.type())}">
-            <p:selectCheckboxMenu id="text-dropdown" dynamic="true" filter="true" styleClass="col-12 m-0 p-0 grid formgrid"
-              filterMatchMode="startsWith" updateLabel="true" scrollHeight="250"  required="true"
-              value="#{processesAnalyticsBean.selectedCustomFilters[fieldEntry.key]}">
-              <f:selectItems value="#{processesAnalyticsBean.customFieldsByType[fieldEntry.key]}" var="optionValue"
+          <h:panelGroup layout="block"
+            rendered="#{managedBean.isStringOrTextCustomFieldType(fieldEntry.key.customFieldMeta.type())}">
+            <p:selectCheckboxMenu id="text-dropdown" dynamic="true" filter="true"
+              styleClass="col-12 m-0 p-0 grid formgrid" filterMatchMode="startsWith" updateLabel="true"
+              scrollHeight="250" required="true" value="#{managedBean.selectedCustomFilters[fieldEntry.key]}">
+              <f:selectItems value="#{managedBean.customFieldsByType[fieldEntry.key]}" var="optionValue"
                 itemValue="#{optionValue}" itemLabel="#{optionValue}" />
               <p:ajax event="change" process="@this" partialSubmit="true" />
             </p:selectCheckboxMenu>
             <p:message styleClass="col-12 m-0 p-2 grid formgrid" for="text-dropdown" />
-            </h:panelGroup>
+          </h:panelGroup>
 
           <!-- TIMESTAMP -->
           <h:panelGroup layout="block">
-          <p:datePicker id="date-range" selectionMode="range" maxdate="#{processesAnalyticsBean.timeIntervalFilter.to}"
-            styleClass="col-12 m-0 p-0 grid formgrid"
-            value="#{processesAnalyticsBean.selectedCustomFilters[fieldEntry.key]}" readonlyInput="true" required="true"
-            requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/utils/process/analyser/ProcessesMonitor/DateErrorMessage')}"
-            rendered="#{processesAnalyticsBean.filterDropdownVisible and fieldEntry.key.customFieldMeta.type() eq 'TIMESTAMP'}">
-            <p:ajax event="dateSelect" process="@this" partialSubmit="true" />
-          </p:datePicker>
-          <p:message styleClass="col-12 m-0 p-2 grid formgrid" for="date-range" />
+            <p:datePicker id="date-range" selectionMode="range" maxdate="#{managedBean.timeIntervalFilter.to}"
+              styleClass="col-12 m-0 p-0 grid formgrid" value="#{managedBean.selectedCustomFilters[fieldEntry.key]}"
+              readonlyInput="true" required="true" pattern="dd/MM/yyyy"
+              requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/utils/process/analyser/ProcessesMonitor/DateErrorMessage')}"
+              rendered="#{'TIMESTAMP' == fieldEntry.key.customFieldMeta.type()}">
```

Comment:
move the render condition to <h:panelgroup> tag at line 75 to make the code consistent with other